Show command aliases in help with setting to disable

PredaaA opened the pull request:
### Type

- [ ] Bugfix
- [x] Enhancement
- [ ] New feature

### Description of the changes
This PR just add back aliases in help formatter, I liked to have them before help rewritte.

PredaaA committed:
Update help.py

PredaaA committed:
Create 3040.enhance.rst

mikeshardmind commented on redbot/core/commands/help.py:
```diff
@@ -187,6 +187,11 @@ def get_default_tagline(ctx: Context):
         tagline = (await ctx.bot._config.help.tagline()) or self.get_default_tagline(ctx)
         signature = f"`Syntax: {ctx.clean_prefix}{command.qualified_name} {command.signature}`"
+        signature += (
+            f"\n`Alias{'es' if len(command.aliases) > 1 else ''}: {', '.join(command.aliases)}`"
```
This construction isn't i18n safe and cannot be used as-is.

mikeshardmind replied:
Ignore this for now, the i18n stuff wasn't fully handled and this can be adapted for i18n when it is.

mikeshardmind commented:
This needs to be temporarily frozen until some of the help pagination behavior is changed in 3.2.1

This is *not* a rejection of adding this.

Jackenmen committed:
remove towncrier entry

Jackenmen commented:
The mentioned changes to the pagination have been done already so I'm unfreezing this.

@PredaaA can you resolve the conflicts and make the construction mentioned by Sinbad compatible with i18n?

PredaaA commented:
> The mentioned changes to the pagination have been done already so I'm unfreezing this.
> 
> @PredaaA can you resolve the conflicts and make the construction mentioned by Sinbad compatible with i18n?

Sure. I will do that later today.

PredaaA committed:
Merge branch 'V3/develop' into patch-4

PredaaA committed:
Make it i18n friendly.

PredaaA committed:
That was uneeded to change this actually.

PredaaA committed:
...

PredaaA committed:
..

PredaaA committed:
.

Jackenmen reviewed (changes requested):
Could you add a setting for showing aliases in help? It can be enabled by default.

PredaaA committed:
Add a setting for aliases.

PredaaA committed:
DOTS

PredaaA commented:
> Could you add a setting for showing aliases in help? It can be enabled by default.

Done.👍

Jackenmen commented on redbot/core/core_commands.py:
```diff
@@ -1439,15 +1439,31 @@ async def helpset_showhidden(self, ctx: commands.Context, show_hidden: bool = No
+        if show_aliases:
+            await ctx.send(_("Help will not show commands aliases."))
+        else:
+            await ctx.send(_("Help will show commands aliases."))
```
These two strings should be the other way around.
```suggestion
        if show_aliases:
            await ctx.send(_("Help will show commands aliases."))
        else:
            await ctx.send(_("Help will not show commands aliases."))
```

Jackenmen commented on redbot/core/commands/help.py:
```diff
@@ -212,6 +213,9 @@ async def format_command_help(
         )
+        if help_settings.show_aliases and command.aliases:
+            alias_fmt = T_("Aliases") if len(command.aliases) > 1 else T_("Alias")
+            signature += f"\n`{alias_fmt}: {humanize_list(command.aliases)}`"
```
This should probably show the prefix before the alias name and it also needs support for command groups (already discussed in Discord, just posting this here for completeness of review)
```suggestion
            signature += f"\n`{alias_fmt}: {humanize_list(command.aliases)}`"
```

Jackenmen replied:
You will need to update handling for non-embed help but as discussed in Discord, this can wait until we decide on the formatting.

Jackenmen replied:
We need some limit for amount of shown aliases as well as their max length.

PredaaA committed:
Update redbot/core/core_commands.py

Co-authored-by: jack1142 <6032823+jack1142@users.noreply.github.com>

PredaaA committed:
Merge branch 'V3/develop' into patch-4

PredaaA committed:
Address requested changes maybe

Jackenmen commented on redbot/core/commands/help.py:
```diff
@@ -209,9 +210,22 @@ async def format_command_help(
-        signature = _(
-            "`Syntax: {ctx.clean_prefix}{command.qualified_name} {command.signature}`"
-        ).format(ctx=ctx, command=command)
+        signature = (
+            f"`{_('Syntax')}: {ctx.clean_prefix}{command.qualified_name} {command.signature}`"
+        )
```
Something went wrong with conflict resolution here - these lines weren't changed in this PR but now they are.

Jackenmen reviewed (changes requested):
There's one issue with conflict resolution + the non-embed help does not work properly with this PR as mentioned in the previous [review comment](https://github.com/Cog-Creators/Red-DiscordBot/pull/3040#discussion_r418624582). I said that it can wait until we decide on the formatting (which we did), so you have 2 options:
1. Update help's formatting as discussed [here (link to discussion in #old-advanced-coding)](https://discordapp.com/channels/133049272517001216/251782448629809152/705816866479669358)
2. Leave the formatting as is and make separate PR for that (assuming you want to do it of course).

And after either of these, you need to handle the non-embed help properly: there shouldn't be backticks in the formatting of non-embed help. Example of how it looks now:
![image](https://user-images.githubusercontent.com/6032823/87067167-f9d6ea00-c213-11ea-8bb0-66fcb682728d.png)

PredaaA committed:
New format + changes requested.

PredaaA committed:
okay we'll get there someday

PredaaA committed:
honk

Drapersniper committed:
aaaaaaaaaaaaaaa

PredaaA committed:
Merge pull request #3 from Drapersniper/wtf

aaaaaaaaaaaaaaa

PredaaA committed:
Black

PredaaA committed:
Merge branch 'V3/develop' into patch-4

PredaaA committed:
Fix missing humanize_timedelta import.

Jackenmen committed:
Merge branch 'V3/develop' into patch-4

Jackenmen committed:
style

Jackenmen committed:
Two things from my very old pending review, see commit desc

```py
            valid_alias_list = [
                af
                for a in aliases
                if (af := f"{a}")
                and len(af) < 500
                and ((a_counter + len(af)) < 500)
                and (a_counter := a_counter + len(af))
            ]
```
^ This can be simplified:
```suggestion
            valid_alias_list = [
                alias
                for alias in aliases
                if (a_counter := a_counter + len(alias)) < 500
            ]
```
Although I think it would be somewhat clearer to use a `for` loop rather than a list comprehension (+ we can just `break` when there's not gonna be another alias that could fit in the list since it's sorted):
```suggestion
            valid_alias_list = []

            for alias in aliases:
                if (a_counter := a_counter + len(alias)) < 500:
                    valid_alias_list.append(alias)
                else:
                    break
```

Jackenmen committed:
style *again*

Jackenmen committed:
use qualified name of the parent command

Jackenmen committed:
meh

Jackenmen committed:
another meh

Jackenmen committed:
Revert the last commit...

Jackenmen reviewed (approved):
It's time.
